bam_to_mate_hist.py

```diff
@@ -254,38 +332,58 @@ def extract_stats(stat_list, bamfile, outfile_name, count_diff_refname_stub=Fals
     Returns:
         stat_dict ({str:float/str}: mappings of the stats and data suitable to be consumed by report generator
     '''
-    stat_dict = {}
+    # some path handling, doesn't need to happen in here really.
     stat_dict["BAM_FILE_PATH"] = os.path.split(bamfile)[-1]
-    num_pairs = stat_list[7]
-    stat_dict["NUM_PAIRS"] = num_pairs
-
-    print "Histograms written to:", os.path.abspath(outfile_name + "_long.png"), os.path.abspath(outfile_name + "_short.png")
     stat_dict["PATH_TO_LONG_HIST"] = os.path.abspath(outfile_name + "_long.png")
     stat_dict["PATH_TO_SHORT_HIST"] = os.path.abspath(outfile_name + "_short.png")
+    print "Histograms written to:", os.path.abspath(outfile_name + "_long.png"), os.path.abspath(outfile_name + "_short.png")
```

Comment:
should probably use the stat_dict["PATH_TO_*_HIST"] values instead of joining the paths again. Not blocking, but generally good practice to not recalculate values if you can

Reply:
good point.